crates/primitives/src/prune/checkpoint.rs

```diff
@@ -0,0 +1,13 @@
+use crate::{prune::PruneMode, BlockNumber};
+use reth_codecs::{main_codec, Compact};
+
+/// Saves the pruning progress of a stage.
+#[main_codec]
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+#[cfg_attr(test, derive(Default))]
+pub struct PruneCheckpoint {
+    /// Highest pruned block number.
+    block_number: BlockNumber,
+    /// Prune mode.
+    prune_mode: PruneMode,
+}
```

Comment:
does this belong in primitives

doesn't hurt here, I think

Reply:
Same as with stage checkpoints. We want to use prune checkpoints in database, pipeline and background task, so it needs to be shared between them. Not sure if there's a better place to put some shared stuff.